src/lib.rs

```diff
@@ -168,11 +216,33 @@ impl zed::Extension for Java {
             env.push(("CLASSPATH".to_string(), classpath));
         }
 
+        let mut args = Vec::new();
+
+        // Add lombok as javaagent if enabled
+        let lombok_enabled = LspSettings::for_worktree(language_server_id.as_ref(), worktree)?
+            .settings
+            .and_then(|settings| {
+                settings
+                    .get("lombok")
+                    .and_then(|lombok| lombok.get("enabled"))
+                    .and_then(|enabled| enabled.as_bool())
+            })
+            .unwrap_or(false);
```

Comment:
Why not check `initialization_options.settings.java.jdt.ls.lombokSupport.enabled` instead of creating a new setting? It is lengthy, but we'll be documenting how to use Lombok anyway, so users won't have to look for this option themselves.

Reply:
And `settings.lombok.version` can be changed to `settings.lombok_version`.

Reply:
Will take a look, thanks!